xnmt/translator.py

```diff
@@ -277,9 +292,23 @@ def __call__(self, translator, dec_state, src, trg):
       except ValueError:
         pass
       # Calculate the evaluation score
-      eval_score.append(self.evaluation_metric.evaluate_fast(trg_i.words, sample_i))
-
-    return -dy.sum_elems(dy.cmult(dy.inputTensor(eval_score, batched=True), dy.esum(logsofts)))
+      score = 0 if not len(sample_i) else self.evaluation_metric.evaluate_fast(trg_i.words, sample_i)
+      self.eval_score.append(score)
+      self.trueScore = dy.inputTensor(self.eval_score, batched=True)
+
+    final_score = -self.trueScore
+    if self.use_baseline:
```

Comment:
I think this is a bit different than I know. The baseline should be predicted at every timestep and be multiplied with the log prob at that time step, right? @neubig

Reply:
Oh sorry, I initially read the algorithm and assumed that the word level predictions would be time-consuming. (I thought that the loss for Linear Regression would be according to BLEU score at every t-1th step which seemed too costly and ambiguous). 
So the intention was to develop the pipeline for sentence level approximation and then get to the word level in the next commit. But it seems to require just overall BLEU Score which makes complete sense and is not slow. Shall change it in next iteration. Thanks.

Reply:
Ah yes.. You are correct. I think this is true if we are using the reward for the whole sequence :) Please pardon my previous comment.

Reply:
Hi!  I guess your initial observation seems to be valid. I might be wrong, but by adding the projected score (the projection of the final BLEU score) at every instance, we might be able to penalize the sentences that deviated early on, more.